# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546488060 28800
#      Wed Jan 02 20:01:00 2019 -0800
# Branch stable
# Node ID a480ccada034bb7e5ce9bfcbdd6ba0e333944134
# Parent  80e98774b2b009ce6973386a961383b688d88e5c
ccolamd.cc: Fix static analyzer detected issues (bug #55347).

* ccolamd.cc (Fcsymamd): Fix incorrect off-by-one checking of input knobs
argument.

diff --git a/libinterp/dldfcn/ccolamd.cc b/libinterp/dldfcn/ccolamd.cc
--- a/libinterp/dldfcn/ccolamd.cc
+++ b/libinterp/dldfcn/ccolamd.cc
@@ -417,19 +417,19 @@ colamd, csymamd, amd, colamd, symamd, an
   // Check for user-passed knobs
   if (nargin > 1)
     {
       NDArray User_knobs = args(1).array_value ();
       int nel_User_knobs = User_knobs.numel ();
 
       if (nel_User_knobs > 0)
         knobs[CCOLAMD_DENSE_ROW] = User_knobs(0);
-      if (nel_User_knobs > 0)
+      if (nel_User_knobs > 1)
         knobs[CCOLAMD_AGGRESSIVE] = User_knobs(1);
-      if (nel_User_knobs > 1)
+      if (nel_User_knobs > 2)
         spumoni = static_cast<int> (User_knobs(2));
 
       // print knob settings if spumoni is set
       if (spumoni)
         {
           octave_stdout << "\ncsymamd version " << CCOLAMD_MAIN_VERSION
                         << '.' << CCOLAMD_SUB_VERSION
                         << ", " << CCOLAMD_DATE << "\n";
